src/models/chan.js

```diff
@@ -80,6 +80,10 @@ Chan.prototype.pushMessage = function(client, msg, increasesUnread) {
 		if (msg.highlight) {
 			this.highlight++;
 		}
+	} else if (msg.self && this.firstUnread > 0) {
```

Comment:
I don't think you need the `this.firstUnread > 0`. You could rewrite this if as `if(msg.self)` and move the if above to just `else if (!isOpen)`

Reply:
Updated. Theory was that it'd be more performant to only update the variable if it needs to be updated, but that's probably me being silly. JS vars probably don't even work like that.